File: common/src/main/java/net/impactdev/gts/common/messaging/interpreters/Interpreter.java
package net.impactdev.gts.common.messaging.interpreters;

import net.impactdev.gts.common.plugin.GTSPlugin;

pulic interface Interpreter {

    void register(GTSPlugin plugin);

    void getDecoders(GTSPlugin plugin);

    void getInterpreters(GTSPlugin plugin);

}

File: common/src/main/kotlin/net/impactdev/gts/common/messaging/interpreters/Interpreter.kt
package net.impactdev.gts.common.messaging.interpreters

import net.impactdev.gts.common.plugin.GTSPlugin

interface Interpreter {
    fun register(plugin: GTSPlugin?)
    fun getDecoders(plugin: GTSPlugin?)
    fun getInterpreters(plugin: GTSPlugin?)
}